src/jltypes.c

```diff
@@ -572,7 +572,43 @@ static int typekey_eq(jl_datatype_t *tt, jl_value_t **key, size_t n)
     return 1;
 }
 
+static int typekeyvalue_eq(jl_datatype_t *tt, jl_value_t *key1, jl_value_t **key, size_t n, int leaf)
```

Comment:
Would be good to have a comment that this is the same as the previous function, but with instances in the keys instead of types.